src/plugins/discover/public/application/view_components/utils/use_search.ts

```diff
@@ -115,24 +115,23 @@ export const useSearch = (services: DiscoverViewServices) => {
     requests: new RequestAdapter(),
   };
 
-  const getDatasetAutoSearchOnPageLoadPreference = () => {
-    // Checks the searchOnpageLoadPreference for the current dataset if not specifed defaults to true
-    const datasetType = data.query.queryString.getQuery().dataset?.type;
-
-    const datasetService = data.query.queryString.getDatasetService();
-
-    return !datasetType || (datasetService?.getType(datasetType)?.meta?.searchOnLoad ?? true);
-  };
-
   const shouldSearchOnPageLoad = useCallback(() => {
+    // Checks the searchOnpageLoadPreference for the current dataset if not specifed defaults to UI Settings
+    const { queryString } = data.query;
+    const { dataset } = queryString.getQuery();
+    const typeConfig = dataset ? queryString.getDatasetService().getType(dataset.type) : undefined;
+    const datasetPreference =
+      typeConfig?.meta?.searchOnLoad ?? uiSettings.get(SEARCH_ON_PAGE_LOAD_SETTING);
+
     // A saved search is created on every page load, so we check the ID to see if we're loading a
     // previously saved search or if it is just transient
     return (
-      services.uiSettings.get(SEARCH_ON_PAGE_LOAD_SETTING) ||
+      datasetPreference ||
+      uiSettings.get(SEARCH_ON_PAGE_LOAD_SETTING) ||
```

Comment:
why repeat this when part of datasetPreference assignment above?

Reply:
+1

Reply:
i believe i was just over thinking. in terms of code it is redundant but i was little bit worried that return true if undefined isn't the most accurate depiction of what should happen.

but with fresh eyes that doesn't really make sense. in terms of the ci taking an hour, could this be a fast follow?